source/common/upstream/subset_lb.h

```diff
@@ -177,8 +198,14 @@ class SubsetLoadBalancer : public LoadBalancer, Logger::Loggable<Logger::Id::ups
   LbSubsetEntryPtr fallback_subset_;
   LbSubsetEntryPtr panic_mode_subset_;
 
+  absl::optional<LbSubsetEntryPtr> selector_fallback_subset_any_;
+  absl::optional<LbSubsetEntryPtr> selector_fallback_subset_default_;
```

Comment:
I think the ownership of the above pointers is never actually shared. It does not hurt for them to be shared_ptr, but it is more readable to use std::unique_ptr to clearly indicate that this member is only owned by the parent SubsetLoadBalancer object.